src/utils.ts

```diff
@@ -614,6 +614,58 @@ async function verifyFlavorPath(flavor: BrowserFlavor | undefined, platform: Pla
     }
 }
 
+type ExtensionSettings = [string, boolean | string | {[key: string]: string} | undefined];
+
+export function reportExtensionSettings(telemetryReporter: Readonly<TelemetryReporter>): void {
+    const extensionSettingsList = Object.entries(vscode.workspace.getConfiguration(SETTINGS_STORE_NAME)).splice(4) as Array<ExtensionSettings>;
+    const settings = vscode.workspace.getConfiguration(SETTINGS_STORE_NAME);
+    const changedSettingsMap: Map<string, string> = new Map<string, string>();
+    for (const currentSetting of extensionSettingsList) {
+        const settingName: string = currentSetting[0];
+        const settingValue: boolean | string | {[key: string]: string} | undefined = currentSetting[1];
+        const settingInspect = settings.inspect(settingName);
+        if (settingInspect) {
+            const defaultValue = settingInspect.defaultValue;
+            if (settingValue !== undefined && settingValue !== defaultValue) {
+                if (defaultValue && typeof defaultValue === 'object' && typeof settingValue === 'object') {
```

Comment:
`object` value settings are kind of tricky.  For our extension, these are the path mapping ones where the values look something like this and is edited directly in `settings.json`:
```
{
    "/": "${workspaceFolder}"
}
```

When trying to change this setting, it actually links you directly to the `settings.json` file like so:
![image](https://user-images.githubusercontent.com/14304598/120729715-b709aa80-c494-11eb-99f0-d91911004890.png)

When retrieving these values, it is placed inside a `Proxy` object so we cannot do direct object comparison, but have to compare property by property.

If there is a difference detected between the user entry and the default entry, we add the changed setting to the map and then break the loop.